packages/components/src/components/Modal/ModalHeader.tsx

```diff
@@ -0,0 +1,42 @@
+import React from "react";
+import { DialogDismiss } from "ariakit/dialog";
+import { Box } from "../../primitives/Box";
+import { Button } from "../Button";
+
+export interface ModalHeaderProps
+  extends Omit<React.HTMLAttributes<HTMLDivElement>, "color"> {
+  children: NonNullable<React.ReactNode>;
+}
+
+const ModalHeader = React.forwardRef<HTMLDivElement, ModalHeaderProps>(
+  ({ children, ...props }, ref) => {
+    return (
+      <Box.div
+        borderBottomColor="colorBorderWeakest"
+        borderBottomStyle="borderSolid"
+        borderBottomWidth="borderWidth10"
+        display="flex"
+        gap="space30"
+        justifyContent="space-between"
+        padding="space60"
+        ref={ref}
+        {...props}
+      >
+        <Box.div>{children}</Box.div>
+        <Box.div>
+          <Button
+            aria-label="Close modal"
+            as={DialogDismiss as unknown as keyof JSX.IntrinsicElements}
+            iconOnly
+            leadingIcon="XMarkIcon"
+            variant="ghost"
+          />
```

Comment:
Instead of doing it this way and having to mess with the cast of DialogDismiss, you can actually pass `Button` to the `as` prop for `DialogDismiss`:

```       
<DialogDismiss
    as={Button}
    aria-label="Close modal"
    iconOnly
    leadingIcon="XMarkIcon"
    variant="ghost"
  />

Reply:
Good catch!